effect/_intents.py

```diff
@@ -54,6 +52,50 @@ def parallel(effects):
     return Effect(ParallelEffects(list(effects)))
 
 
+@attributes(['exception', 'index'])
+class FirstError(Exception):
```

Comment:
I just realized that this isn't publicly exposed anywhere. This should probably be exposed as effect.FirstError?